nekton/nii2dcm.py

```diff
@@ -0,0 +1,157 @@
+
+        writer = pydicom_seg.MultiClassWriter(
+            template=seg_map,
+            inplane_cropping=False,  # Crop image slices to the minimum bounding box on
+            # x and y axes. Maybe not supported by other frameworks.
+            skip_empty_slices=True,  # Don't encode slices with only zeros
+            skip_missing_segment=True,  # If a segment definition is missing in the
+            # template, then raise an error instead of
+            # skipping it.
+        )
+
+        # write the image
+        dcmseg = writer.write(segImage_itk, source_images=[dcmImage])
+        # correct the acquition time and other info if neccesary
+        dcmseg.AcquisitionTime = dcmImage.AcquisitionTime
+        if appendedImagePosition:
+            dcmseg.ImagePositionPatient = None
+
+        return dcmseg
+
+    def multilabel_converter(
+        self, segfiles=List[Path], dcmfiles=List[Path]
+    ) -> List[Path]:
+        # all_individual_segs = [nib.load(file).get_fdata() for file in segfiles]
+        # de-priorisited for now
+
+        raise NotImplementedError(
+            "Multilabel NifTi s to DICOMSeg cannot to be handled yet!!"
+        )
+
+    def multiclass_converter(
+        self, segfile: Path, segMapping: Path, dcmfiles: List[Path]
+    ) -> List[Path]:
+        """Convert a given nifti segmentation to dicomseg for multiclass labels
+
+        Args:
+            segfile (Path): path to the nifti segmentation file
+            segMapping (Path): path to the dcmqii format segmentation mapping json
+            dcmfiles (List[Path]): list of paths of all the source dicom files
+
+        Returns:
+            List[Path]: list of paths of all generated dicomseg files
+        """
+        # load the segmentation mapping
+        seg_map = self._load_segmap(segMapping)
+        # load the segmentation and verify if all dicoms exist
+        seg = nib.load(segfile).get_fdata()
+        sorted_dcmfiles = self._check_all_dicoms(dcmfiles, seg)
+
+        # create folder to store the dicomsegs
+        parent_dir = Path(segfile).parent
+        out_folder = os.path.join(parent_dir, "dicomseg")
+        os.makedirs(out_folder, exist_ok=True)
+
+        # list to contain all the output paths of the dicomseg created
+
+        out_list = []
+
+        # create store individual dicomseg
+        for i in range(len(sorted_dcmfiles)):
+
+            # for non-empty segmenetation only
+            non_zero_labels = np.unique(seg[..., i : i + 1]) != 0  # noqa
+            if sum(non_zero_labels) > 0:
+                dcm_file = Path(sorted_dcmfiles[i])
+                dcm = pydicom.read_file(dcm_file)
+                dcmseg = self._create_dicomseg(
+                    seg_map, seg[..., i : i + 1], dcm  # noqa
+                )
+                out_dcmfile = Path(os.path.join(out_folder, dcm_file.name))
+                dcmseg.save_as(out_dcmfile)
+                out_list.append(out_dcmfile)
```

Comment:
is it not possible to have a multilayer dicom seg, instead of multiple files?

Reply:
added support for multiplayer dicomseg